src/main/webapp/js/instructorHome.js

```diff
@@ -1,27 +1,61 @@
+var COURSE_PANELS_TO_AUTO_LOAD_COUNT = 3;
+
 $(document).ready(function() {
     setupFsCopyModal();
     
     //Click event binding for radio buttons
-    var radiobuttons = $("label[name='sortby']");
-    $.each(radiobuttons, function() {
+    var radioButtons = $('label[name="sortby"]');
+    $.each(radioButtons, function() {
         $(this).click(function () {
             var currentPath = window.location.pathname;
             var query = window.location.search.substring(1);
             var params = {};
             
-            var param_values = query.split("&");
-            for (var i = 0; i < param_values.length; i++) {
-                var param_value = param_values[i].split("=");
-                params[param_value[0]] = param_value[1];
+            var paramValues = query.split('&');
+            for (var i = 0; i < paramValues.length; i++) {
+                var paramValue = paramValues[i].split('=');
+                params[paramValue[0]] = paramValue[1];
             }
 
-            if ("user" in params == false) {
-                params["user"] = $("input[name='user']").val();
+            if ('user' in params == false) {
+                params['user'] = $('input[name="user"]').val();
             }
 
-            window.location.href = currentPath + "?user=" + params["user"] + "&sortby=" + $(this).attr("data");
+            window.location.href = currentPath + '?user=' + params['user'] + '&sortby=' + $(this).attr('data');
+        });
+    });
+    
+    // AJAX loading of course panels
+    var $coursePanels = $('div[id|="course"]');
+    $.each($coursePanels, function() {
+        $(this).filter(function() {
+            var isNotLoaded = $(this).find('form').length;
+            return isNotLoaded;
+        }).click(function() {
+            var $panel = $(this);
+            var formData = $panel.find('form').serialize();
+            var content = $panel.find('.pull-right')[0];
+            
+            $.ajax({
+                type : 'POST',
+                url : '/page/instructorHomePage?' + formData,
+                beforeSend : function() {
+                    $(content).html("<img src='/images/ajax-loader.gif'/>");
+                },
+                error : function() {
+                    // TODO: show an error message?         
+                },
+                success : function(data) {
+                    // .replaceWith() fixes <br> in statuses' tooltips by closing the tooltip <span>
```

Comment:
It's not meant for the reviewer.
It's meant to explain why it's using `$panel[0].outerHTML = data;` instead of a more conventional `$panel.replaceWith(data)` in case anyone plans on refactoring it in the future.
I suppose it could be made more clear, though.